spring-kafka/src/main/java/org/springframework/kafka/event/ListenerContainerIdleEvent.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.event;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.kafka.common.TopicPartition;
+
+/**
+ * An event that is emitted when a container is idle if the container
+ * is configured to do so.
+ *
+ * @author Gary Russell
+ *
+ */
+@SuppressWarnings("serial")
+public class ListenerContainerIdleEvent extends KafkaEvent {
+
+	private final long idleTime;
+
+	private final String listenerId;
+
+	private final List<TopicPartition> topicPartitions;
+
+	public ListenerContainerIdleEvent(Object source, long idleTime, String id,
+			Collection<TopicPartition> topicPartitions) {
+		super(source);
+		this.idleTime = idleTime;
+		this.listenerId = id;
+		this.topicPartitions = new ArrayList<TopicPartition>(topicPartitions);
+	}
+
+	/**
+	 * How long the container has been idle.
+	 * @return the time in milliseconds.
+	 */
+	public long getIdleTime() {
+		return this.idleTime;
+	}
+
+	/**
+	 * The queues the container is listening to.
+	 * @return the queue names.
+	 */
+	public String[] getTopicPartitions() {
+		return this.topicPartitions.toArray(new String[this.topicPartitions.size()]);
```

Comment:
Can't we require exactly this value as a ctor arg? 
To avoid `TopicPartition` API dependency.
Not sure that it will work everywhere, because, I think we may end up with the ClassNotFoundException anyway on the `source`, if the event listener doesn't have appropriate classes in its CP.